src/Form/BaseSettingsForm.php

```diff
@@ -0,0 +1,263 @@
+      '#size' => 30,
+    ];
+
+    // Add the Debug Configuration.
+    $form['enable_logging'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable FB Instant Articles SDK logging?'),
+      '#default_value' => $config->get('enable_logging'),
+      '#description' => $this->t('Sends Facebook Instant Articles SDK logging messages to Drupal watchdog.'),
+    ];
+
+    // Add the Canonical URL override.
+    $form['canonical_url_override'] = [
+      '#type' => 'textfield',
+      '#title' => t('Canonical URL override'),
+      '#default_value' => $config->get('canonical_url_override', ''),
+      '#description' => t('If you need to override the base url portion of the canonical URL, you can do so here. This may be helpful for development domains or necessary if admin users perform tasks that trigger Facebook requests from alternate domains. This URL should not include a trailing slash (e.g. http://drupal.org).'),
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * Validate the Ad Source Url field.
+   *
+   * @param array $form
+   *   FAPI array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state.
+   */
+  public function validateAdSourceUrl(array &$form, FormStateInterface $form_state) {
+    // Only validate if Source URL is selected as ad type.
+    if ($form_state->getValue('ads_type') != FB_INSTANT_ARTICLES_AD_TYPE_SOURCE_URL) {
+      return;
+    }
+
+    $ads_iframe_url = $form_state->getValue('ads_iframe_url');
+    if (empty($ads_iframe_url)) {
+      $form_state->setErrorByName('ads_iframe_url', $this->t('You must specify a valid source URL for your Ads when using the Source URL ad type.'));
+    }
+
+    if (filter_var($ads_iframe_url, FILTER_VALIDATE_URL) === FALSE) {
+      $form_state->setErrorByName('ads_iframe_url', $this->t('You must specify a valid source URL for your Ads when using the Source URL ad type.'));
+    }
+  }
+
+  /**
+   * Validate the Audience Network placement id.
+   *
+   * @param array $form
+   *   FAPI array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state.
+   */
+  public function validateAnPlacementId(array &$form, FormStateInterface $form_state) {
+    // Only validate if Audience Network is selected as ad type.
+    if ($form_state->getValue('ads_type') != FB_INSTANT_ARTICLES_AD_TYPE_FBAN) {
+      return;
+    }
+
+    $ads_an_placement_id = $form_state->getValue('ads_an_placement_id');
+    if (empty($ads_an_placement_id)) {
+      $form_state->setErrorByName('ads_an_placement_id', $this->t('You must specify a valid Placement ID when using the Audience Network ad type.'));
+    }
+
+    if (preg_match('/^[\d_]+$/', $ads_an_placement_id) !== 1) {
+      $form_state->setErrorByName('ads_an_placement_id', $this->t('You must specify a valid Placement ID when using the Audience Network ad type.'));
```

Comment:
It looks like this is something set up in the brand's Audience Network account for Instant Articles.  Adding the instrux from the FB developer docs to the form feedback.